core/lib/zksync_core_leftovers/src/lib.rs

```diff
@@ -741,6 +748,41 @@ pub async fn initialize_components(
         .context("add_tee_verifier_input_producer_to_task_futures()")?;
     }
 
+    if components.contains(&Component::DADispatcher) {
+        if eth
+            .sender
+            .clone()
+            .context("eth_sender")?
+            .pubdata_sending_mode
+            != PubdataSendingMode::Custom
+        {
+            // Warning instead of returning an error is appropriate here because the DA dispatcher
+            // is not a critical component. It's more convenient for tests and local setup to enable
+            // it by default, but don't start the component if the pubdata sending mode is not `Custom`
+            tracing::warn!("DA dispatcher is enabled, but the pubdata sending mode is not `Custom`. DA dispatcher will not be started.");
+        } else {
+            let started_at = Instant::now();
+            let da_config = configs
+                .da_dispatcher_config
+                .clone()
+                .context("da_dispatcher_config")?;
+            let da_dispatcher_pool =
+                ConnectionPool::<Core>::singleton(database_secrets.master_url()?)
```

Comment:
Given that there's concurrency inside the DA dispatcher, shouldn't the DB pool have 2 connections at least? The same question applies to initializing the DA dispatcher using the node framework.